compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -1087,4 +1087,211 @@ mod tests {
             assert_eq!(got, token);
         }
     }
+
+    // returns a vector of:
+    //   (expected_token_discriminator, strings_to_lex)
+    // expected_token_discriminator matches a given token when
+    // std::mem::discriminant returns the same discriminant for both.
+    fn blns_base64_to_statements(base64_str: String) -> Vec<(Option<Token>, Vec<String>)> {
+        use base64::engine::general_purpose;
+        use std::borrow::Cow;
+        use std::io::Cursor;
+        use std::io::Read;
+
+        let mut wrapped_reader = Cursor::new(base64_str);
+        let mut decoder =
+            base64::read::DecoderReader::new(&mut wrapped_reader, &general_purpose::STANDARD);
+        let mut base64_decoded = Vec::new();
+        decoder.read_to_end(&mut base64_decoded).unwrap();
+
+        // NOTE: when successful, this is the same conversion method as used in
+        // noirc_driver::stdlib::stdlib_paths_with_source, viz.
+        //
+        // let source = std::str::from_utf8(..).unwrap().to_string();
+        let s: Cow<'_, str> = match std::str::from_utf8(&base64_decoded) {
+            Ok(s) => std::borrow::Cow::Borrowed(s),
+            Err(_err) => {
+                // recover as much of the string as possible
+                // when str::from_utf8 fails
+                String::from_utf8_lossy(&base64_decoded)
+            }
+        };
+
+        vec![
+            // Token::Ident(_)
+            (
+                None,
+                vec![format!(
+                    "
+                    let \"{}\" = ();
+                ",
+                    s
+                )],
+            ),
+            (
+                Some(Token::Str("".to_string())),
+                vec![format!(
+                    "
+                    let s = \"{}\";
+                ",
+                    s
+                )],
+            ),
+            (
+                Some(Token::RawStr("".to_string(), 0)),
+                vec![
+                    // let s = r"Hello world";
+                    format!(
+                        "
+                    let s = r\"{}\";
+                ",
+                        s
+                    ),
+                    // let s = r#"Simon says "hello world""#;
+                    format!(
+                        "
+                    let s = r#\"{}\"#;
+                ",
+                        s
+                    ),
```

Comment:
Some weird formatting going on here. Can it be fixed/condensed?

Reply:
yes, I used `{s}` instead and it's more condensed after `cargo fmt / clippy`